### PR TITLE
feat(carousel): adds image carousels for Teams page

### LINKED ISSUE #42: create carousel component
screen shot 2019-01-14 at 11 14 14 am

on Teams page

### PR DESCRIPTION
**Pull request by jnm2377**
Closes #42 

Suuuper hacky at the moment, mostly because Gatsby images did not like being displayed either `inline` or `flex`, which I had to do to translate the carousel left/right, so I ended up creating new `img` utilizing the children data (open to feedback on that!). Otherwise functionality is there, radio button nav is keyboard accessible thru arrow keys. Wasn't sure if this was also supposed to be on a timer or anything (is not atm).

Also (not the way code is written atm but...) when I tried to utilize `.map` to iterate through `RadioButtons` depending on the number of images, it kept returning undefined. Unsure why, so at the moment it is hard coded for just 4 imgs/4 radio btns. 

#### Changelog

**New**

- Carousel component

**Changed**

- Teams markdown

### CODE REVIEW

**Comment by alisonjoseph** [comment]
Overall looks good 🎉Some things we should try do (if we have time)

- auto pick number of images, what happens if someone adds 3, does it break?
- sync up with design on auto-play or not
- look at possibly using gatsby-image this way, not a deal breaker https://www.gatsbyjs.org/packages/gatsby-image/ `import Img from "gatsby-image"`

File: src/components/Carousel/Carousel.js [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+    const img = children[1].props.children[3].props;
+    const img2 = children[3].props.children[3].props;
+    const img3 = children[5].props.children[3].props;
+    const img4 = children[7].props.children[3].props;
+
+    return (
+      <div className="ibm--carousel">
+        <div className="ibm--carousel-slide-wrapper">
+          <div className={`ibm--carousel-slide ${this.props.id}`}>
+            <img src={img.src} alt={img.alt} className={img.className} />
```
Comment (alisonjoseph) [inline src/components/Carousel/Carousel.js]: we need alt tags

**Comment by RyanCaruthers** [comment]
Any chance we can size down the dots overall, and what about it switching to the square layout on mobile reflected in the sketch file? is that piece not doable?

screen shot 2019-01-24 at 1 17 54 pm

**Comment by alisonjoseph** [comment]
What pixel size for the dots?

And this component is specific for just images, won't work for the cards.

**Comment by RyanCaruthers** [comment]
8px size dots and 8px of space between each

File: src/components/Carousel/Carousel.js (hunk 1 above)
Comment (alisonjoseph) [reply src/components/Carousel/Carousel.js]: ha nvm. Ignore me

**Comment by alisonjoseph** [comment]
oh square, not cards 🤦‍♀️  We could possibly switch to a square on mobile, just would have to figure out how to crop the images, or add something for different mobile images. Might be more work than its worth right now.

**Comment by jnm2377** [comment]
> * auto pick number of images, what happens if someone adds 3, does it break?

@alisonjoseph, this was another reason I thought I should have a count prop or something similar. Am definitely trying to come up with a fix for this. I have a few ideas to try out.

> * look at possibly using gatsby-image this way, not a deal breaker https://www.gatsbyjs.org/packages/gatsby-image/ `import Img from "gatsby-image"`

will look into it!

Re: 
> * Any chance we can size down the dots overall, and what about it switching to the square layout on mobile reflected in the sketch file? is that piece not doable?
> * oh square, not cards 🤦‍♀️ We could possibly switch to a square on mobile, just would have to figure out how to crop the images, or add something for different mobile images. Might be more work than its worth right now.

@RyanCaruthers @alisonjoseph One solution might be containing each individual image in a div and making it take up the whole div (similar to like what background-image/background-size cover would do), and then changing the div width and height for mobile view. 🤷‍♀️ I would still have to figure out how to adjust the translation stuff for mobile view since they won't be taking up 100% of the slide width, and the onChange function translates -100% each time.

**Comment by RyanCaruthers** [comment]
totally open to easier solutions that still look great!

**Comment by RyanCaruthers** [comment]
Ah, all good @jnm2377 things are looking awesome, keep it up in the order you need to! thanks!

**Comment by RyanCaruthers** [comment]
@jnm2377 @alisonjoseph Is auto carousel something we could add to this request? Apologies, I didn't even think about it before, but wanted to make the ask and see.

**Comment by RyanCaruthers** [comment]
@jnm2377 If the "auto Carousel" part is too taxing at this point, please don't add that function to the carousel at this point. Thanks!

**Comment by jnm2377** [comment]
@RyanCaruthers the auto carousel is probably easier than the touch stuff for mobile, I can focus on the touch and if I have time to add the auto carousel at the end, I can do that.

**Comment by RyanCaruthers** [comment]
sounds good @jnm2377 !

**Comment by jnm2377** [comment]
@RyanCaruthers the autoplay is included now! And should still update if a user clicks the radio button/swipes on mobile view. Still need make the changes that crop the image on mobile view though, if we're still wanting those. @alisonjoseph I'm still working on cleaning up the code/make it more dynamic to allow for any given number of images, but feel free to leave any comments.

**Comment by RyanCaruthers** [comment]
Once this is merged. Scott and I will update the images within each carousel tomorrow. FYI

**Comment by RyanCaruthers** [comment]
Ah ok all good yall! and sweet on the autoplay! ok keep going! sorry to jump the gun. lol excited

**Comment by RyanCaruthers** [comment]
@jnm2377 Auto play is great!!!

**Comment by alisonjoseph** [comment]
Hey @RyanCaruthers thoughts on making the clickable area bigger than the dots? I feel like we might have some guidelines around this somewhere. 

This looks great!

**Comment by jnm2377** [comment]
> Hey @RyanCaruthers thoughts on making the clickable area bigger than the dots? I feel like we might have some guidelines around this somewhere.

@alisonjoseph @RyanCaruthers I thought that might be an issue as well. Just added draggable functionality for desktop. One thing I noticed though was that since I have it auto rotate images every 6 seconds, say I swipe to the next image at second 5, it'll swipe again in 1 second. I need to figure out how to reset the countdown after each swipe, so will probably refactor that today.

**Comment by RyanCaruthers** [comment]
@jnm2377 @alisonjoseph Hmmm i like the smaller size aesthetically, I say keep it for now, I know Brand may even desire the dots to be even closer to each other... but I say roll with this. The larger dots are not tasteful.

**Comment by jnm2377** [comment]
@RyanCaruthers I added a mouse event for swipeable functionality on the images in in case the radio buttons are too small. The netlify link above doesn't seem to be updating for me. @alisonjoseph were you able to see it? My local env is different than what the Netlify link shows, so it might not be updated for you, Ryan.

**Comment by RyanCaruthers** [comment]
@jnm2377 sounds great! Sadek from brand team is reviewing now to get final confirmation on dots size and layout. coming soon!

**Comment by alisonjoseph** [comment]
Is it updating now @jnm2377 ? I checked netlify and all the builds passed. might have to clear cache/private window.

**Comment by RyanCaruthers** [comment]
ok gang! @jnm2377 @alisonjoseph final tweak then we are good to merge this one!

Lets scoot the dots closer together now... Dots should be 8px and 8px apart from each other. 
See dots at bottom of the attached image....

screen shot 2019-01-31 at 12 27 47 pm

**Comment by alisonjoseph** [comment]
One minor thing I know Mike will notice. can we align image with text on the right?
screen shot 2019-01-31 at 12 50 59 pm

**Comment by alisonjoseph** [comment]
nevermind. it is aligned

**Comment by jnm2377** [comment]
Still trying to solve rendering images dynamically so don't merge yet! 😬 @RyanCaruthers @alisonjoseph

**Comment by alisonjoseph** [comment]
ok 👍wasn't sure if you were doing that now or later

**Comment by alisonjoseph** [comment]
Yea, lets get this merged. and come back to it. there are a couple merge conflicts.

**Pull request merged**